h2/src/test/org/h2/test/db/TestCsv.java

```diff
@@ -634,9 +644,81 @@ private void testReadEmptyNumbers2() throws Exception {
         Connection conn = DriverManager.getConnection("jdbc:h2:mem:test");
         Statement stat = conn.createStatement();
         stat.execute("CREATE TABLE TEST(TEST DECIMAL(12,2) NULL)");
-        stat.execute("INSERT INTO TEST SELECT * FROM CsvRead('" + fileName + "')");
+        stat.execute("INSERT INTO TEST SELECT * FROM CsvRead('" + fileName + "', NULL, 'quotedNulls=true')");
 
         FileUtils.delete(fileName);
     }
 
+    private void testCsvQuotedString1() throws Exception { testCsvQuotedNullStrings(false, "NULL"); }
+    private void testCsvQuotedString2() throws Exception { testCsvQuotedNullStrings(true, "NULL"); }
+    private void testCsvQuotedString3() throws Exception { testCsvQuotedNullStrings(false, ""); }
+    private void testCsvQuotedString4() throws Exception { testCsvQuotedNullStrings(true, ""); }
+    private void testCsvQuotedString5() throws Exception { testCsvQuotedNullStrings(false, "$empty"); }
+    private void testCsvQuotedString6() throws Exception { testCsvQuotedNullStrings(true, "$empty"); }
+    
+    private void testCsvQuotedNullStrings(boolean quotedStrings, String nullString) throws Exception {
+        String fileName = getBaseDir() + "/test.csv";
+        File csvFile = new File(fileName);
+        if (csvFile.exists()) {
+            csvFile.delete();
+        }
+        
+        deleteDb("csv");
+        Connection conn = DriverManager.getConnection("jdbc:h2:mem:test");
+        Statement stat = conn.createStatement();
+        stat.execute("DROP TABLE IF EXISTS TEST");
+        stat.execute("CREATE TABLE TEST(ID char(2) NOT NULL, NAME varchar(255), HEIGHT integer, BIRTHDATE date, PRIMARY KEY (ID))");
+        stat.execute("INSERT INTO TEST VALUES('01', 'Penrosed Roberto', 511, '1958-03-29')");
+        stat.execute("INSERT INTO TEST VALUES('02', NULL, 512, '1975-07-12')");
+        stat.execute("INSERT INTO TEST VALUES('03', 'Smith John', NULL, '1971-11-03')");
+        stat.execute("INSERT INTO TEST VALUES('04', 'Hatchet Eve', 500, NULL)");
+        stat.execute("INSERT INTO TEST VALUES('05', NULL, NULL, NULL)");
+        stat.execute("CALL CSVWRITE('" + fileName + "', 'SELECT * FROM TEST ORDER BY ID','quotedNulls=" + quotedStrings + " nullString=" + nullString + "')");
+        
+        FileInputStream fis = new FileInputStream(csvFile);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int read = fis.read(buffer);
+        while(read>=0) {
```

Comment:
This file also has minor problems with formatting here and below, please use the same code style as the whole project has.

Reply:
Fixed.